{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "MdgTZ9fLN68q",
        "outputId": "9f1759e5-6c6f-451c-d905-f72f32b01395"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  blends-common default-jre default-jre-headless fonts-dejavu-core\n",
            "  fonts-dejavu-extra libapache-pom-java libargs4j-java libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libcommons-compress-java libcommons-io-java\n",
            "  libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl\n",
            "  libgail-common libgail18 libgtk2.0-0 libgtk2.0-bin libgtk2.0-common libhts3\n",
            "  libhtscodecs2 libhtsjdk-java libjbzip2-java libjson-simple-java libkdf5-2\n",
            "  libncbi-vdb2 libncbi-wvdb2 libngs-java libngs-sdk-dev libngs-sdk2\n",
            "  librsvg2-common libsis-base-java libsis-base-jni libsis-jhdf5-java\n",
            "  libsis-jhdf5-jni libsnappy-java libsnappy-jni libxtst6 libxxf86dga1\n",
            "  med-config menu openjdk-11-jre openjdk-8-jre-headless x11-utils\n",
            "Suggested packages:\n",
            "  python3-numpy python3-matplotlib texlive-latex-recommended blends-doc\n",
            "  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc\n",
            "  libavalon-framework-java libcommons-logging-java-doc\n",
            "  libexcalibur-logkit-java liblog4j1.2-java gvfs picard-tools\n",
            "  libjson-simple-doc menu-l10n gksu | kde-runtime | ktsuss libnss-mdns\n",
            "  fonts-nanum fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei\n",
            "  fonts-wqy-zenhei fonts-indic cwltool mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  bcftools blends-common bwa default-jre default-jre-headless fastqc\n",
            "  fonts-dejavu-core fonts-dejavu-extra libapache-pom-java libargs4j-java\n",
            "  libatk-wrapper-java libatk-wrapper-java-jni libcommons-compress-java\n",
            "  libcommons-io-java libcommons-jexl2-java libcommons-lang3-java\n",
            "  libcommons-logging-java libcommons-math3-java libcommons-parent-java\n",
            "  libfindbin-libs-perl libgail-common libgail18 libgtk2.0-0 libgtk2.0-bin\n",
            "  libgtk2.0-common libhts3 libhtscodecs2 libhtsjdk-java libjbzip2-java\n",
            "  libjson-simple-java libkdf5-2 libncbi-vdb2 libncbi-wvdb2 libngs-java\n",
            "  libngs-sdk-dev libngs-sdk2 librsvg2-common libsis-base-java libsis-base-jni\n",
            "  libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxtst6\n",
            "  libxxf86dga1 med-config menu openjdk-11-jre openjdk-8-jre\n",
            "  openjdk-8-jre-headless samtools sra-toolkit trimmomatic x11-utils\n",
            "0 upgraded, 54 newly installed, 0 to remove and 35 not upgraded.\n",
            "Need to get 55.8 MB of archives.\n",
            "After this operation, 165 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bwa amd64 0.7.17-6 [195 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.27+6~us1-0ubuntu1~22.04 [214 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:38 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:39 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Get:40 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libgtk2.0-common all 2.24.33-2ubuntu2.1 [125 kB]\n",
            "Get:41 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libgtk2.0-0 amd64 2.24.33-2ubuntu2.1 [2,038 kB]\n",
            "Get:42 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libgail18 amd64 2.24.33-2ubuntu2.1 [15.9 kB]\n",
            "Get:43 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libgail-common amd64 2.24.33-2ubuntu2.1 [132 kB]\n",
            "Get:44 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libgtk2.0-bin amd64 2.24.33-2ubuntu2.1 [7,936 B]\n",
            "Get:45 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:46 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:47 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:48 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 librsvg2-common amd64 2.52.5+dfsg-3ubuntu0.2 [17.7 kB]\n",
            "Get:49 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-8-jre-headless amd64 8u452-ga~us1-0ubuntu1~22.04 [30.8 MB]\n",
            "Get:50 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-8-jre amd64 8u452-ga~us1-0ubuntu1~22.04 [75.3 kB]\n",
            "Get:51 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Get:52 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:53 http://archive.ubuntu.com/ubuntu jammy/universe amd64 trimmomatic all 0.39+dfsg-2 [758 kB]\n",
            "Get:54 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 55.8 MB in 8s (6,820 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 126319 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../01-libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package bcftools.\n",
            "Preparing to unpack .../02-bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Selecting previously unselected package menu.\n",
            "Preparing to unpack .../03-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../04-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package bwa.\n",
            "Preparing to unpack .../05-bwa_0.7.17-6_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-6) ...\n",
            "Selecting previously unselected package default-jre-headless.\n",
            "Preparing to unpack .../06-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre-headless (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../07-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package openjdk-11-jre:amd64.\n",
            "Preparing to unpack .../08-openjdk-11-jre_11.0.27+6~us1-0ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-11-jre:amd64 (11.0.27+6~us1-0ubuntu1~22.04) ...\n",
            "Selecting previously unselected package default-jre.\n",
            "Preparing to unpack .../09-default-jre_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "Preparing to unpack .../10-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../11-libjbzip2-java_0.9.1-7_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-7) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../12-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../13-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../14-libcommons-compress-java_1.21-1_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.21-1) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../15-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../16-libcommons-jexl2-java_2.1.1-5_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-5) ...\n",
            "Selecting previously unselected package libjson-simple-java.\n",
            "Preparing to unpack .../17-libjson-simple-java_2.3.0-1_all.deb ...\n",
            "Unpacking libjson-simple-java (2.3.0-1) ...\n",
            "Selecting previously unselected package libngs-sdk2:amd64.\n",
            "Preparing to unpack .../18-libngs-sdk2_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev:amd64.\n",
            "Preparing to unpack .../19-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-java:amd64.\n",
            "Preparing to unpack .../20-libngs-java_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-java:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../21-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../22-libsnappy-java_1.1.8.3-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../23-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Selecting previously unselected package libfindbin-libs-perl.\n",
            "Preparing to unpack .../24-libfindbin-libs-perl_2.190.02-1_all.deb ...\n",
            "Unpacking libfindbin-libs-perl (2.190.02-1) ...\n",
            "Selecting previously unselected package libargs4j-java.\n",
            "Preparing to unpack .../25-libargs4j-java_2.33-1.1_all.deb ...\n",
            "Unpacking libargs4j-java (2.33-1.1) ...\n",
            "Selecting previously unselected package libcommons-io-java.\n",
            "Preparing to unpack .../26-libcommons-io-java_2.11.0-2_all.deb ...\n",
            "Unpacking libcommons-io-java (2.11.0-2) ...\n",
            "Selecting previously unselected package libcommons-lang3-java.\n",
            "Preparing to unpack .../27-libcommons-lang3-java_3.11-1_all.deb ...\n",
            "Unpacking libcommons-lang3-java (3.11-1) ...\n",
            "Selecting previously unselected package libsis-base-jni.\n",
            "Preparing to unpack .../28-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n",
            "Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-base-java.\n",
            "Preparing to unpack .../29-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n",
            "Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-jhdf5-jni.\n",
            "Preparing to unpack .../30-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n",
            "Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package libsis-jhdf5-java.\n",
            "Preparing to unpack .../31-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n",
            "Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../32-fastqc_0.11.9+dfsg-5_all.deb ...\n",
            "Unpacking fastqc (0.11.9+dfsg-5) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../33-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../34-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../35-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../36-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../37-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../38-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libgtk2.0-common.\n",
            "Preparing to unpack .../39-libgtk2.0-common_2.24.33-2ubuntu2.1_all.deb ...\n",
            "Unpacking libgtk2.0-common (2.24.33-2ubuntu2.1) ...\n",
            "Selecting previously unselected package libgtk2.0-0:amd64.\n",
            "Preparing to unpack .../40-libgtk2.0-0_2.24.33-2ubuntu2.1_amd64.deb ...\n",
            "Unpacking libgtk2.0-0:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Selecting previously unselected package libgail18:amd64.\n",
            "Preparing to unpack .../41-libgail18_2.24.33-2ubuntu2.1_amd64.deb ...\n",
            "Unpacking libgail18:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Selecting previously unselected package libgail-common:amd64.\n",
            "Preparing to unpack .../42-libgail-common_2.24.33-2ubuntu2.1_amd64.deb ...\n",
            "Unpacking libgail-common:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Selecting previously unselected package libgtk2.0-bin.\n",
            "Preparing to unpack .../43-libgtk2.0-bin_2.24.33-2ubuntu2.1_amd64.deb ...\n",
            "Unpacking libgtk2.0-bin (2.24.33-2ubuntu2.1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../44-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../45-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../46-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package librsvg2-common:amd64.\n",
            "Preparing to unpack .../47-librsvg2-common_2.52.5+dfsg-3ubuntu0.2_amd64.deb ...\n",
            "Unpacking librsvg2-common:amd64 (2.52.5+dfsg-3ubuntu0.2) ...\n",
            "Selecting previously unselected package openjdk-8-jre-headless:amd64.\n",
            "Preparing to unpack .../48-openjdk-8-jre-headless_8u452-ga~us1-0ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-8-jre-headless:amd64 (8u452-ga~us1-0ubuntu1~22.04) ...\n",
            "Selecting previously unselected package openjdk-8-jre:amd64.\n",
            "Preparing to unpack .../49-openjdk-8-jre_8u452-ga~us1-0ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-8-jre:amd64 (8u452-ga~us1-0ubuntu1~22.04) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../50-samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../51-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package trimmomatic.\n",
            "Preparing to unpack .../52-trimmomatic_0.39+dfsg-2_all.deb ...\n",
            "Unpacking trimmomatic (0.39+dfsg-2) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../53-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up default-jre-headless (2:1.11-72build2) ...\n",
            "Setting up bwa (0.7.17-6) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up openjdk-11-jre:amd64 (11.0.27+6~us1-0ubuntu1~22.04) ...\n",
            "Setting up libfindbin-libs-perl (2.190.02-1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up default-jre (2:1.11-72build2) ...\n",
            "Setting up libjson-simple-java (2.3.0-1) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up openjdk-8-jre-headless:amd64 (8u452-ga~us1-0ubuntu1~22.04) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/orbd to provide /usr/bin/orbd (orbd) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/servertool to provide /usr/bin/servertool (servertool) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/tnameserv to provide /usr/bin/tnameserv (tnameserv) in auto mode\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libjbzip2-java (0.9.1-7) ...\n",
            "Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Setting up librsvg2-common:amd64 (2.52.5+dfsg-3ubuntu0.2) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Setting up libgtk2.0-common (2.24.33-2ubuntu2.1) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libngs-java:amd64 (2.11.2-3) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up libsnappy-jni (1.1.8.3-1) ...\n",
            "Setting up libargs4j-java (2.33-1.1) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up trimmomatic (0.39+dfsg-2) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Setting up libgtk2.0-0:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-5) ...\n",
            "Setting up libcommons-lang3-java (3.11-1) ...\n",
            "Setting up libgail18:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Setting up libgtk2.0-bin (2.24.33-2ubuntu2.1) ...\n",
            "Setting up libsnappy-java (1.1.8.3-1) ...\n",
            "Setting up libcommons-compress-java (1.21-1) ...\n",
            "Setting up libgail-common:amd64 (2.24.33-2ubuntu2.1) ...\n",
            "Setting up libcommons-io-java (2.11.0-2) ...\n",
            "Setting up openjdk-8-jre:amd64 (8u452-ga~us1-0ubuntu1~22.04) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/policytool to provide /usr/bin/policytool (policytool) in auto mode\n",
            "Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Setting up fastqc (0.11.9+dfsg-5) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libgdk-pixbuf-2.0-0:amd64 (2.42.8+dfsg-1ubuntu0.3) ...\n",
            "Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y sra-toolkit fastqc trimmomatic bwa samtools bcftools openjdk-8-jre"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\n",
        "!unzip Trimmomatic-0.39.zip\n",
        "!ln -s /content/Trimmomatic-0.39/trimmomatic-0.39.jar /usr/local/bin/trimmomatic.jar"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JeLsppGVVZ3p",
        "outputId": "4fe486aa-5c2a-44a2-a651-68dbf9aabd2b"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-06-19 17:56:21--  http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\n",
            "Resolving www.usadellab.org (www.usadellab.org)... 80.83.125.111\n",
            "Connecting to www.usadellab.org (www.usadellab.org)|80.83.125.111|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 133596 (130K) [application/zip]\n",
            "Saving to: ‘Trimmomatic-0.39.zip’\n",
            "\n",
            "Trimmomatic-0.39.zi 100%[===================>] 130.46K   208KB/s    in 0.6s    \n",
            "\n",
            "2025-06-19 17:56:23 (208 KB/s) - ‘Trimmomatic-0.39.zip’ saved [133596/133596]\n",
            "\n",
            "Archive:  Trimmomatic-0.39.zip\n",
            "   creating: Trimmomatic-0.39/\n",
            "  inflating: Trimmomatic-0.39/LICENSE  \n",
            "  inflating: Trimmomatic-0.39/trimmomatic-0.39.jar  \n",
            "   creating: Trimmomatic-0.39/adapters/\n",
            "  inflating: Trimmomatic-0.39/adapters/NexteraPE-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq2-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq2-SE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-PE-2.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-SE.fa  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!prefetch SRR12905311\n",
        "!fasterq-dump SRR12905311 --split-files"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xMjIRj_LVf9K",
        "outputId": "f7910b3c-0f7c-4622-dd30-29f8b50298d9"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "2025-06-19T17:58:20 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-06-19T17:58:21 prefetch.2.11.3: 1) Downloading 'SRR12905311'...\n",
            "2025-06-19T17:58:21 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2025-06-19T17:58:21 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2025-06-19T17:59:11 prefetch.2.11.3:  HTTPS download succeed\n",
            "2025-06-19T17:59:13 prefetch.2.11.3:  'SRR12905311' is valid\n",
            "2025-06-19T17:59:13 prefetch.2.11.3: 1) 'SRR12905311' was downloaded successfully\n",
            "2025-06-19T17:59:13 prefetch.2.11.3: 'SRR12905311' has 0 unresolved dependencies\n",
            "spots read      : 6,371,441\n",
            "reads read      : 12,742,882\n",
            "reads written   : 12,742,882\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Исследуется влияние меди на устойчивость к антибиотикам у резистентных и чувствительных бактерий P.aeruginosa"
      ],
      "metadata": {
        "id": "tt-rj_IUXCT6"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/765/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
        "!gunzip GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
        "!mv GCF_000006765.1_ASM676v1_genomic.fna PAO1_ref.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hP6-WswTXmCy",
        "outputId": "90c4d611-e07f-42c5-ef84-887468b35d09"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-06-19 18:12:41--  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/765/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.31, 130.14.250.7, 130.14.250.10, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1787220 (1.7M) [application/x-gzip]\n",
            "Saving to: ‘GCF_000006765.1_ASM676v1_genomic.fna.gz’\n",
            "\n",
            "GCF_000006765.1_ASM 100%[===================>]   1.70M  1.38MB/s    in 1.2s    \n",
            "\n",
            "2025-06-19 18:12:44 (1.38 MB/s) - ‘GCF_000006765.1_ASM676v1_genomic.fna.gz’ saved [1787220/1787220]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc SRR12905311_1.fastq SRR12905311_2.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2EpbUm0rZOzK",
        "outputId": "54585e8f-a221-404d-8b38-8bf206edf5b9"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of SRR12905311_1.fastq\n",
            "Approx 5% complete for SRR12905311_1.fastq\n",
            "Approx 10% complete for SRR12905311_1.fastq\n",
            "Approx 15% complete for SRR12905311_1.fastq\n",
            "Approx 20% complete for SRR12905311_1.fastq\n",
            "Approx 25% complete for SRR12905311_1.fastq\n",
            "Approx 30% complete for SRR12905311_1.fastq\n",
            "Approx 35% complete for SRR12905311_1.fastq\n",
            "Approx 40% complete for SRR12905311_1.fastq\n",
            "Approx 45% complete for SRR12905311_1.fastq\n",
            "Approx 50% complete for SRR12905311_1.fastq\n",
            "Approx 55% complete for SRR12905311_1.fastq\n",
            "Approx 60% complete for SRR12905311_1.fastq\n",
            "Approx 65% complete for SRR12905311_1.fastq\n",
            "Approx 70% complete for SRR12905311_1.fastq\n",
            "Approx 75% complete for SRR12905311_1.fastq\n",
            "Approx 80% complete for SRR12905311_1.fastq\n",
            "Approx 85% complete for SRR12905311_1.fastq\n",
            "Approx 90% complete for SRR12905311_1.fastq\n",
            "Approx 95% complete for SRR12905311_1.fastq\n",
            "Analysis complete for SRR12905311_1.fastq\n",
            "Started analysis of SRR12905311_2.fastq\n",
            "Approx 5% complete for SRR12905311_2.fastq\n",
            "Approx 10% complete for SRR12905311_2.fastq\n",
            "Approx 15% complete for SRR12905311_2.fastq\n",
            "Approx 20% complete for SRR12905311_2.fastq\n",
            "Approx 25% complete for SRR12905311_2.fastq\n",
            "Approx 30% complete for SRR12905311_2.fastq\n",
            "Approx 35% complete for SRR12905311_2.fastq\n",
            "Approx 40% complete for SRR12905311_2.fastq\n",
            "Approx 45% complete for SRR12905311_2.fastq\n",
            "Approx 50% complete for SRR12905311_2.fastq\n",
            "Approx 55% complete for SRR12905311_2.fastq\n",
            "Approx 60% complete for SRR12905311_2.fastq\n",
            "Approx 65% complete for SRR12905311_2.fastq\n",
            "Approx 70% complete for SRR12905311_2.fastq\n",
            "Approx 75% complete for SRR12905311_2.fastq\n",
            "Approx 80% complete for SRR12905311_2.fastq\n",
            "Approx 85% complete for SRR12905311_2.fastq\n",
            "Approx 90% complete for SRR12905311_2.fastq\n",
            "Approx 95% complete for SRR12905311_2.fastq\n",
            "Analysis complete for SRR12905311_2.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar Trimmomatic-0.39/trimmomatic-0.39.jar PE -phred33 \\\n",
        "SRR12905311_1.fastq SRR12905311_2.fastq \\\n",
        "SRR12905311_1_trimmed.fastq SRR12905311_1_unpaired.fastq \\\n",
        "SRR12905311_2_trimmed.fastq SRR12905311_2_unpaired.fastq \\\n",
        "ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HZeNXyLZZpxS",
        "outputId": "5fd2d384-41a4-48f7-9086-b82004c0c6a1"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 SRR12905311_1.fastq SRR12905311_2.fastq SRR12905311_1_trimmed.fastq SRR12905311_1_unpaired.fastq SRR12905311_2_trimmed.fastq SRR12905311_2_unpaired.fastq ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Multiple cores found: Using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-PE.fa (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticPE.run(TrimmomaticPE.java:552)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:80)\n",
            "Input Read Pairs: 6371441 Both Surviving: 6195847 (97.24%) Forward Only Surviving: 128758 (2.02%) Reverse Only Surviving: 37336 (0.59%) Dropped: 9500 (0.15%)\n",
            "TrimmomaticPE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa index PAO1_ref.fa\n",
        "!bwa mem -t 2 PAO1_ref.fa SRR12905311_1_trimmed.fastq SRR12905311_2_trimmed.fastq > aligned.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "mzNxBN1QawiK",
        "outputId": "e1777217-5639-4e7d-f38a-0e9897f5bded"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bwa_index] Pack FASTA... 0.04 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 2.85 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.04 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.03 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 1.59 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index PAO1_ref.fa\n",
            "[main] Real time: 4.707 sec; CPU: 4.550 sec\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 253438 sequences (20000027 bp)...\n",
            "[M::process] read 253394 sequences (20000045 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (455, 104798, 14, 456)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 183, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
            "[M::mem_pestat] mean and std.dev: (187.82, 80.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.76, 84.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (73, 401, 1524)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4426)\n",
            "[M::mem_pestat] mean and std.dev: (755.08, 988.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5877)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 187, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (190.98, 82.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253438 reads in 18.007 CPU sec, 10.115 real sec\n",
            "[M::process] read 253346 sequences (20000139 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (470, 104847, 19, 483)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 181, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 471)\n",
            "[M::mem_pestat] mean and std.dev: (185.34, 84.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 586)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (302.20, 84.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (365, 1430, 4655)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13235)\n",
            "[M::mem_pestat] mean and std.dev: (2134.05, 2335.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17525)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 173, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (183.07, 77.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253394 reads in 17.730 CPU sec, 11.391 real sec\n",
            "[M::process] read 253334 sequences (20000141 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (510, 104921, 11, 473)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 176, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (181.59, 78.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 554)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (303.19, 84.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 222, 1457)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4129)\n",
            "[M::mem_pestat] mean and std.dev: (692.55, 731.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5465)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 176, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (180.69, 80.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253346 reads in 18.653 CPU sec, 12.667 real sec\n",
            "[M::process] read 253436 sequences (20000011 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (498, 104938, 14, 494)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 177, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (184.37, 80.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.10, 84.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (269, 569, 1443)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3791)\n",
            "[M::mem_pestat] mean and std.dev: (693.62, 542.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4965)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (141, 193, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 438)\n",
            "[M::mem_pestat] mean and std.dev: (193.54, 75.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 537)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253334 reads in 17.788 CPU sec, 10.992 real sec\n",
            "[M::process] read 253208 sequences (20000008 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (503, 104878, 20, 486)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 180, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 488)\n",
            "[M::mem_pestat] mean and std.dev: (188.12, 82.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.54, 84.67)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (932, 1968, 6015)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16181)\n",
            "[M::mem_pestat] mean and std.dev: (2966.50, 2723.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21264)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 189, 256)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 504)\n",
            "[M::mem_pestat] mean and std.dev: (193.29, 85.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 628)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253436 reads in 17.163 CPU sec, 10.149 real sec\n",
            "[M::process] read 253534 sequences (20000140 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (521, 104816, 19, 505)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 173, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (181.48, 80.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.73, 84.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (284, 501, 1892)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5108)\n",
            "[M::mem_pestat] mean and std.dev: (919.06, 973.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6716)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 184, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (190.91, 85.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253208 reads in 18.479 CPU sec, 12.691 real sec\n",
            "[M::process] read 253242 sequences (20000094 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (473, 104887, 14, 480)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 183, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (187.47, 80.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (301.93, 84.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (140, 442, 1369)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3827)\n",
            "[M::mem_pestat] mean and std.dev: (784.50, 949.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5056)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 174, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (180.57, 81.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253534 reads in 17.296 CPU sec, 11.098 real sec\n",
            "[M::process] read 253494 sequences (20000073 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (502, 104609, 15, 515)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 187, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (185.77, 80.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.42, 84.67)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (412, 1001, 1932)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4972)\n",
            "[M::mem_pestat] mean and std.dev: (995.85, 720.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6492)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 173, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 457)\n",
            "[M::mem_pestat] mean and std.dev: (181.34, 82.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253242 reads in 16.623 CPU sec, 9.532 real sec\n",
            "[M::process] read 253262 sequences (20000151 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (554, 104607, 7, 467)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 189, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
            "[M::mem_pestat] mean and std.dev: (185.89, 81.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.82, 84.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 183, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 488)\n",
            "[M::mem_pestat] mean and std.dev: (188.86, 83.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253494 reads in 17.448 CPU sec, 12.090 real sec\n",
            "[M::process] read 253556 sequences (20000074 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (521, 104890, 15, 495)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 174, 229)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
            "[M::mem_pestat] mean and std.dev: (175.16, 76.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.35, 84.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (273, 944, 1361)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3537)\n",
            "[M::mem_pestat] mean and std.dev: (782.29, 587.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4625)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 190, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (189.42, 82.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253262 reads in 16.784 CPU sec, 10.161 real sec\n",
            "[M::process] read 253228 sequences (20000061 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (473, 104764, 22, 493)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 187, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (194.53, 85.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 603)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (302.85, 84.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (352, 1458, 2042)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5422)\n",
            "[M::mem_pestat] mean and std.dev: (1633.77, 1644.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8212)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 178, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (180.94, 78.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 552)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253556 reads in 16.364 CPU sec, 10.322 real sec\n",
            "[M::process] read 253482 sequences (20000014 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (564, 104663, 14, 489)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 182, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (188.62, 80.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.27, 84.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (28, 423, 1451)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4297)\n",
            "[M::mem_pestat] mean and std.dev: (651.38, 710.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5720)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 175, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 498)\n",
            "[M::mem_pestat] mean and std.dev: (187.49, 85.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 623)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253228 reads in 17.703 CPU sec, 12.293 real sec\n",
            "[M::process] read 253060 sequences (20000002 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (532, 104930, 11, 484)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 186, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (187.78, 83.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (301.43, 84.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (548, 1295, 4629)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12791)\n",
            "[M::mem_pestat] mean and std.dev: (2633.00, 3292.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16872)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 178, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (183.24, 82.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253482 reads in 15.763 CPU sec, 8.397 real sec\n",
            "[M::process] read 253376 sequences (20000071 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (524, 104746, 21, 491)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 181, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (185.32, 83.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 585)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.73, 84.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (185, 733, 1998)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5624)\n",
            "[M::mem_pestat] mean and std.dev: (971.56, 1252.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7437)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 178, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (182.32, 80.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 613)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253060 reads in 17.584 CPU sec, 12.207 real sec\n",
            "[M::process] read 253082 sequences (20000126 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (499, 104521, 22, 474)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 184, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 494)\n",
            "[M::mem_pestat] mean and std.dev: (188.85, 84.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 617)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.81, 84.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 1046, 1392)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3682)\n",
            "[M::mem_pestat] mean and std.dev: (765.32, 545.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4827)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 181, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 479)\n",
            "[M::mem_pestat] mean and std.dev: (181.03, 80.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253376 reads in 17.539 CPU sec, 12.309 real sec\n",
            "[M::process] read 253428 sequences (20000114 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (490, 104984, 13, 468)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (114, 177, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 501)\n",
            "[M::mem_pestat] mean and std.dev: (182.08, 88.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 630)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (301.91, 84.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (187, 322, 1501)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4129)\n",
            "[M::mem_pestat] mean and std.dev: (517.36, 489.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5443)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 176, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (179.02, 81.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253082 reads in 16.588 CPU sec, 8.624 real sec\n",
            "[M::process] read 253156 sequences (20000059 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (512, 104464, 20, 488)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 174, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
            "[M::mem_pestat] mean and std.dev: (179.66, 77.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 539)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.66, 84.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 439, 1436)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4060)\n",
            "[M::mem_pestat] mean and std.dev: (464.53, 482.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5372)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 185, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (189.30, 84.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 603)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253428 reads in 17.407 CPU sec, 12.407 real sec\n",
            "[M::process] read 253542 sequences (20000055 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (508, 104828, 17, 516)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 180, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (182.32, 79.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.67, 84.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (223, 520, 1399)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3751)\n",
            "[M::mem_pestat] mean and std.dev: (651.62, 454.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4927)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 177, 253)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 523)\n",
            "[M::mem_pestat] mean and std.dev: (186.12, 89.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 658)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253156 reads in 16.546 CPU sec, 11.012 real sec\n",
            "[M::process] read 253226 sequences (20000033 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (453, 104657, 14, 465)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 181, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (185.69, 80.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.85, 84.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (281, 769, 911)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2171)\n",
            "[M::mem_pestat] mean and std.dev: (592.46, 399.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2801)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 183, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (189.67, 88.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253542 reads in 15.535 CPU sec, 8.780 real sec\n",
            "[M::process] read 253510 sequences (20000016 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (529, 104866, 18, 446)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 182, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
            "[M::mem_pestat] mean and std.dev: (185.13, 81.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 560)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.00, 84.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (154, 962, 1471)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4105)\n",
            "[M::mem_pestat] mean and std.dev: (935.19, 956.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5422)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 176, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (181.29, 81.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 586)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253226 reads in 17.499 CPU sec, 12.416 real sec\n",
            "[M::process] read 253144 sequences (20000108 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (551, 104565, 23, 499)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 177, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (183.50, 77.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.19, 84.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (164, 529, 1474)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4094)\n",
            "[M::mem_pestat] mean and std.dev: (658.75, 578.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5404)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 186, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (191.79, 86.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253510 reads in 16.995 CPU sec, 9.517 real sec\n",
            "[M::process] read 253440 sequences (20000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (482, 104943, 13, 476)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 185, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 433)\n",
            "[M::mem_pestat] mean and std.dev: (185.36, 75.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (301.14, 84.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (110, 1006, 1905)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5495)\n",
            "[M::mem_pestat] mean and std.dev: (1171.62, 1340.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7290)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 190, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (188.39, 79.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 567)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253144 reads in 17.454 CPU sec, 11.858 real sec\n",
            "[M::process] read 253066 sequences (20000124 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (515, 104471, 28, 487)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 181, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 493)\n",
            "[M::mem_pestat] mean and std.dev: (190.19, 85.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 615)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.57, 84.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (216, 930, 1414)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3810)\n",
            "[M::mem_pestat] mean and std.dev: (742.21, 576.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5008)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 178, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 443)\n",
            "[M::mem_pestat] mean and std.dev: (180.76, 76.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253440 reads in 18.255 CPU sec, 12.761 real sec\n",
            "[M::process] read 253164 sequences (20000098 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (502, 104561, 20, 491)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 173, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 459)\n",
            "[M::mem_pestat] mean and std.dev: (179.22, 84.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.69, 84.74)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (175, 910, 1468)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4054)\n",
            "[M::mem_pestat] mean and std.dev: (741.28, 625.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5347)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 187, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (186.67, 83.57)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253066 reads in 17.484 CPU sec, 9.454 real sec\n",
            "[M::process] read 252980 sequences (20000094 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (493, 104463, 10, 473)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 181, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 471)\n",
            "[M::mem_pestat] mean and std.dev: (187.10, 84.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 586)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.15, 84.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (897, 1509, 1853)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3765)\n",
            "[M::mem_pestat] mean and std.dev: (1087.88, 606.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4721)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 179, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (186.21, 81.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 592)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253164 reads in 17.212 CPU sec, 11.807 real sec\n",
            "[M::process] read 253184 sequences (20000117 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (465, 105001, 10, 448)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 182, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 442)\n",
            "[M::mem_pestat] mean and std.dev: (181.33, 77.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 548)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.62, 84.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (165, 523, 1902)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5376)\n",
            "[M::mem_pestat] mean and std.dev: (953.89, 1157.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7113)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 185, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (189.09, 84.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 597)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252980 reads in 17.226 CPU sec, 12.237 real sec\n",
            "[M::process] read 252968 sequences (20000081 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (513, 104458, 17, 464)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
            "[M::mem_pestat] mean and std.dev: (186.73, 84.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.15, 85.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (347, 726, 1488)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3770)\n",
            "[M::mem_pestat] mean and std.dev: (876.35, 650.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4911)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 183, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (186.49, 76.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 575)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253184 reads in 15.947 CPU sec, 8.230 real sec\n",
            "[M::process] read 253138 sequences (20000000 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (487, 104943, 16, 477)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 176, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (180.73, 78.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 566)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.62, 84.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 1348, 4587)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13513)\n",
            "[M::mem_pestat] mean and std.dev: (2499.94, 2968.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17976)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 181, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 441)\n",
            "[M::mem_pestat] mean and std.dev: (183.47, 75.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252968 reads in 17.287 CPU sec, 12.347 real sec\n",
            "[M::process] read 253080 sequences (20000009 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (487, 104860, 15, 489)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (138, 185, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (189.58, 78.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.32, 84.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (108, 689, 1929)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5571)\n",
            "[M::mem_pestat] mean and std.dev: (1046.54, 1388.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7392)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (186.25, 86.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253138 reads in 16.313 CPU sec, 9.929 real sec\n",
            "[M::process] read 253174 sequences (20000032 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (508, 104939, 20, 556)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 184, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
            "[M::mem_pestat] mean and std.dev: (187.26, 81.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 559)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.12, 84.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (137, 1420, 2019)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5783)\n",
            "[M::mem_pestat] mean and std.dev: (1259.84, 1297.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7665)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 186, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (189.18, 81.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253080 reads in 16.398 CPU sec, 10.427 real sec\n",
            "[M::process] read 253124 sequences (20000025 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (477, 104910, 16, 473)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 178, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
            "[M::mem_pestat] mean and std.dev: (188.31, 86.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.37, 84.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (64, 527, 1965)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5767)\n",
            "[M::mem_pestat] mean and std.dev: (954.40, 1217.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7668)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 180, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (182.47, 75.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253174 reads in 17.240 CPU sec, 12.405 real sec\n",
            "[M::process] read 253324 sequences (20000018 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (491, 104815, 11, 480)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 180, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 478)\n",
            "[M::mem_pestat] mean and std.dev: (188.83, 81.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 594)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.62, 84.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 400, 1326)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3714)\n",
            "[M::mem_pestat] mean and std.dev: (599.45, 583.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4908)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 182, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 489)\n",
            "[M::mem_pestat] mean and std.dev: (186.02, 82.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253124 reads in 16.062 CPU sec, 8.302 real sec\n",
            "[M::process] read 252980 sequences (20000049 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (473, 104761, 22, 469)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 184, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 491)\n",
            "[M::mem_pestat] mean and std.dev: (189.20, 81.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.15, 84.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 496, 1485)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4189)\n",
            "[M::mem_pestat] mean and std.dev: (839.32, 777.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5541)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 181, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (189.50, 80.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253324 reads in 16.931 CPU sec, 12.278 real sec\n",
            "[M::process] read 253220 sequences (20000006 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (497, 104817, 21, 482)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 175, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (180.94, 80.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.14, 84.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (94, 307, 1484)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4264)\n",
            "[M::mem_pestat] mean and std.dev: (608.53, 704.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5654)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 172, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (183.89, 82.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 565)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252980 reads in 15.748 CPU sec, 9.803 real sec\n",
            "[M::process] read 253120 sequences (20000007 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (507, 104596, 16, 489)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 178, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (188.98, 83.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.72, 84.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (261, 910, 1432)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3774)\n",
            "[M::mem_pestat] mean and std.dev: (781.53, 659.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4945)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 181, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 477)\n",
            "[M::mem_pestat] mean and std.dev: (183.52, 79.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253220 reads in 16.432 CPU sec, 10.417 real sec\n",
            "[M::process] read 253226 sequences (20000080 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (537, 104901, 18, 476)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 175, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (179.67, 84.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 579)\n",
            "[M::mem_pestat] mean and std.dev: (300.47, 84.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (249, 348, 607)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1323)\n",
            "[M::mem_pestat] mean and std.dev: (337.07, 200.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1681)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 182, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
            "[M::mem_pestat] mean and std.dev: (183.47, 79.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253120 reads in 17.220 CPU sec, 12.473 real sec\n",
            "[M::process] read 252858 sequences (20000125 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (487, 104606, 22, 496)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 185, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (187.71, 84.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.76, 84.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (273, 1354, 3340)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9474)\n",
            "[M::mem_pestat] mean and std.dev: (1945.50, 1959.02)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12541)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 183, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 487)\n",
            "[M::mem_pestat] mean and std.dev: (189.40, 85.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253226 reads in 14.757 CPU sec, 7.621 real sec\n",
            "[M::process] read 253122 sequences (20000056 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (513, 104478, 22, 535)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (137, 182, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
            "[M::mem_pestat] mean and std.dev: (188.95, 77.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.09, 84.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (148, 999, 1410)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3934)\n",
            "[M::mem_pestat] mean and std.dev: (931.05, 781.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5196)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 182, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 479)\n",
            "[M::mem_pestat] mean and std.dev: (188.20, 82.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 596)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252858 reads in 17.252 CPU sec, 12.552 real sec\n",
            "[M::process] read 252880 sequences (20000057 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (475, 104630, 13, 482)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 175, 223)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 419)\n",
            "[M::mem_pestat] mean and std.dev: (175.79, 73.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 517)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.82, 85.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 972, 2286)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6304)\n",
            "[M::mem_pestat] mean and std.dev: (957.83, 839.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8313)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 176, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (184.51, 81.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253122 reads in 15.927 CPU sec, 9.486 real sec\n",
            "[M::process] read 253060 sequences (20000007 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (468, 104580, 23, 495)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
            "[M::mem_pestat] mean and std.dev: (178.82, 73.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 560)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (302.83, 84.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (265, 951, 1500)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3970)\n",
            "[M::mem_pestat] mean and std.dev: (711.47, 583.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5205)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 182, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (185.85, 80.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252880 reads in 16.746 CPU sec, 11.114 real sec\n",
            "[M::process] read 252792 sequences (20000052 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (536, 104670, 26, 468)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 187, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (185.79, 78.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.34, 84.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (433, 999, 1982)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5080)\n",
            "[M::mem_pestat] mean and std.dev: (1018.68, 816.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6629)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 179, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (181.67, 80.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253060 reads in 17.472 CPU sec, 12.751 real sec\n",
            "[M::process] read 253126 sequences (20000013 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (509, 104612, 19, 489)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 184, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
            "[M::mem_pestat] mean and std.dev: (188.78, 82.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 614)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.83, 84.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (364, 1072, 2033)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5371)\n",
            "[M::mem_pestat] mean and std.dev: (1242.00, 1270.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7040)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 178, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (185.65, 82.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 592)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252792 reads in 15.521 CPU sec, 8.034 real sec\n",
            "[M::process] read 252880 sequences (20000139 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (528, 104672, 27, 499)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 184, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 493)\n",
            "[M::mem_pestat] mean and std.dev: (187.91, 83.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 616)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.53, 84.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (927, 1372, 1845)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3681)\n",
            "[M::mem_pestat] mean and std.dev: (962.27, 644.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4599)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 185, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (186.36, 82.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253126 reads in 17.514 CPU sec, 12.821 real sec\n",
            "[M::process] read 253156 sequences (20000004 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (507, 104237, 15, 515)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 181, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (182.03, 77.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.94, 84.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 194, 1352)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3802)\n",
            "[M::mem_pestat] mean and std.dev: (646.07, 711.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5027)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 187, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (187.46, 79.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252880 reads in 16.585 CPU sec, 9.951 real sec\n",
            "[M::process] read 252926 sequences (20000004 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (555, 104691, 24, 536)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 185, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 479)\n",
            "[M::mem_pestat] mean and std.dev: (193.98, 86.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 594)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.79, 84.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (287, 995, 1927)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5207)\n",
            "[M::mem_pestat] mean and std.dev: (962.10, 1117.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6847)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 181, 253)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 507)\n",
            "[M::mem_pestat] mean and std.dev: (188.38, 83.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 634)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253156 reads in 16.464 CPU sec, 10.885 real sec\n",
            "[M::process] read 253178 sequences (20000129 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (503, 104608, 20, 515)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 180, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (187.03, 81.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (301.45, 84.61)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (109, 599, 1493)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4261)\n",
            "[M::mem_pestat] mean and std.dev: (783.37, 808.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5645)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 182, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 481)\n",
            "[M::mem_pestat] mean and std.dev: (187.13, 84.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252926 reads in 17.389 CPU sec, 12.764 real sec\n",
            "[M::process] read 252834 sequences (20000013 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (509, 104819, 15, 485)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 190, 251)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 485)\n",
            "[M::mem_pestat] mean and std.dev: (194.66, 84.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.81, 84.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (343, 746, 1045)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2449)\n",
            "[M::mem_pestat] mean and std.dev: (707.43, 525.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3151)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 177, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 457)\n",
            "[M::mem_pestat] mean and std.dev: (185.78, 82.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 566)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 253178 reads in 16.129 CPU sec, 8.341 real sec\n",
            "[M::process] read 238216 sequences (18827024 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (515, 104844, 17, 515)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 181, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (180.93, 77.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.36, 85.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (85, 275, 2068)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6034)\n",
            "[M::mem_pestat] mean and std.dev: (973.06, 1374.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8017)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 178, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
            "[M::mem_pestat] mean and std.dev: (180.50, 81.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 252834 reads in 17.352 CPU sec, 12.877 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (469, 98380, 22, 458)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 187, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (191.87, 86.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (301.84, 84.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (458, 1366, 2041)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5207)\n",
            "[M::mem_pestat] mean and std.dev: (1208.32, 867.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6790)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 185, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 471)\n",
            "[M::mem_pestat] mean and std.dev: (186.96, 82.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 586)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 238216 reads in 14.497 CPU sec, 7.807 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem -t 2 PAO1_ref.fa SRR12905311_1_trimmed.fastq SRR12905311_2_trimmed.fastq\n",
            "[main] Real time: 531.772 sec; CPU: 828.815 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -bS aligned.sam > aligned.bam\n",
        "!samtools sort aligned.bam -o aligned_sorted.bam\n",
        "!samtools index aligned_sorted.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "9wux-AipdQEz",
        "outputId": "38b4b86c-7d39-4d13-ba1e-ab896c81618b"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools faidx PAO1_ref.fa\n",
        "!bcftools mpileup -Ou -f PAO1_ref.fa aligned_sorted.bam | \\\n",
        "bcftools call -vmO z --ploidy 1 -o variants.vcf.gz\n",
        "!bcftools index variants.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-qxbrFmJej7k",
        "outputId": "cef8d1e3-b6fd-4472-b7d6-be8cc7a658d9"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[mpileup] 1 samples in 1 input files\n",
            "[mpileup] maximum number of reads per input file set to -d 250\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools view -v snps variants.vcf.gz -i '%QUAL>=20' -Oz -o filtered_snps.vcf.gz\n",
        "!bcftools index filtered_snps.vcf.gz"
      ],
      "metadata": {
        "id": "belSXAEShD2T"
      },
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools stats filtered_snps.vcf.gz > stats.txt\n",
        "!head -n 20 stats.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "C8CXn0HKhH9a",
        "outputId": "0af96488-16fc-40fb-c9b7-dbe237415976"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "# This file was produced by bcftools stats (1.13+htslib-1.13+ds) and can be plotted using plot-vcfstats.\n",
            "# The command line was:\tbcftools stats  filtered_snps.vcf.gz\n",
            "#\n",
            "# Definition of sets:\n",
            "# ID\t[2]id\t[3]tab-separated file names\n",
            "ID\t0\tfiltered_snps.vcf.gz\n",
            "# SN, Summary numbers:\n",
            "#   number of records   .. number of data rows in the VCF\n",
            "#   number of no-ALTs   .. reference-only sites, ALT is either \".\" or identical to REF\n",
            "#   number of SNPs      .. number of rows with a SNP\n",
            "#   number of MNPs      .. number of rows with a MNP, such as CC>TT\n",
            "#   number of indels    .. number of rows with an indel\n",
            "#   number of others    .. number of rows with other type, for example a symbolic allele or\n",
            "#                          a complex substitution, such as ACT>TCGA\n",
            "#   number of multiallelic sites     .. number of rows with multiple alternate alleles\n",
            "#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs\n",
            "# \n",
            "#   Note that rows containing multiple types will be counted multiple times, in each\n",
            "#   counter. For example, a row with a SNP and an indel increments both the SNP and\n",
            "#   the indel counter.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Высокий показатель Ts/Tv ratio (~3.24):Это свидетельствует о значительном преобладании транзиций над трансверсиями. Такой высокий коэффициент является нормальным для бактериальных геномов и указывает на высокую степень консервативности изменений.  \n",
        "Доминирование определенных транзиций:Преобладающие типы транзиций включают замену аденина на гуанин (A→G) и тимина на цитозин (T→C). Эти изменения отражают стабильный характер эволюции бактериальной ДНК, поскольку такие замены менее склонны вызывать значительные функциональные нарушения.  \n",
        "Редкость трансверсий:Трансверсии, такие как A→C и T→G, наблюдаются значительно реже. Такие замены чаще вызывают существенные изменения в структуре белка, что делает их редкими вследствие негативного отбора.  \n",
        "Значение высокой доли транзиций:Высокая доля транзиций связана с меньшим количеством вредных эффектов на белки и геномы, позволяя бактериям сохранять стабильность своего генома. Таким образом, этот феномен отражает адаптивную стратегию бактерий, минимизирующую риск негативных последствий мутаций.​\n",
        "\n",
        "Анализ данных показывает значительную долю транзиций (особенно A→G и T→C), что характерно для большинства бактериальных геномов. Высокий Ts/Tv ratio подтверждает тенденцию к сохранению стабильности генома, несмотря на происходящие мутации. Этот механизм обеспечивает эффективную адаптацию бактерий к различным условиям окружающей среды."
      ],
      "metadata": {
        "id": "KlR4DaH-lsA7"
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "SnqymqNglv7T"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}